src/AstromFit.cc

```diff
@@ -947,295 +947,298 @@ static void write_vect_in_fits(const Eigen::VectorXd &V, const string &FitsName)
+    LSDerivatives(tList, grad);
     clock_t tend = clock();
-    std::cout << "INFO: CPU for J*Jt "
-	      << float(tend-tstart)/float(CLOCKS_PER_SEC) << std::endl;
+    _LastNTrip = tList.size();
+
+    cout << " INFO: End of triplet filling, ntrip = " << tList.size()
+         << " CPU = " << float(tend - tstart) / float(CLOCKS_PER_SEC)
+         << endl;
+
+    SpMat hessian;
+    {
+#if (TRIPLET_INTERNAL_COORD == COL)
+        SpMat jacobian(_nParTot, tList.NextFreeIndex());
+        jacobian.setFromTriplets(tList.begin(), tList.end());
+        // release memory shrink_to_fit is C++11
+        tList.clear(); //tList.shrink_to_fit();
+        clock_t tstart = clock();
+        hessian = jacobian * jacobian.transpose();
+        clock_t tend = clock();
+        std::cout << "INFO: CPU for J*Jt "
+                  << float(tend - tstart) / float(CLOCKS_PER_SEC) << std::endl;
 
 #else
-    SpMat jacobian(tList.NextRank(), _nParTot);
-    jacobian.setFromTriplets(tList.begin(), tList.end());
-    // release memory shrink_to_fit is C++11
-    tList.clear(); //tList.shrink_to_fit();
-    cout << " starting H=JtJ " << endl;
-    hessian = jacobian.transpose()*jacobian;
+        SpMat jacobian(tList.NextRank(), _nParTot);
+        jacobian.setFromTriplets(tList.begin(), tList.end());
+        // release memory shrink_to_fit is C++11
+        tList.clear(); //tList.shrink_to_fit();
+        cout << " starting H=JtJ " << endl;
+        hessian = jacobian.transpose() * jacobian;
 #endif
-  }// release the Jacobian
+    }// release the Jacobian
 
 
-  cout << "INFO: hessian : dim=" << hessian.rows()
-       << " nnz=" << hessian.nonZeros()
-       << " filling-frac = " << hessian.nonZeros()/sqr(hessian.rows()) << endl;
-  cout << "INFO: starting factorization" << endl;
+    cout << "INFO: hessian : dim=" << hessian.rows()
+         << " nnz=" << hessian.nonZeros()
+         << " filling-frac = " << hessian.nonZeros() / sqr(hessian.rows()) << endl;
+    cout << "INFO: starting factorization" << endl;
 
-  tstart = clock();
-  CholmodSimplicialLDLT2<SpMat> chol(hessian);
-  if (chol.info() != Eigen::Success)
+    tstart = clock();
+    CholmodSimplicialLDLT2<SpMat> chol(hessian);
+    if (chol.info() != Eigen::Success)
     {
-      cout << "ERROR: AstromFit::Minimize : factorization failed " << endl;
-      return 2;
+        cout << "ERROR: AstromFit::Minimize : factorization failed " << endl;
+        return 2;
     }
 
-  tend = clock();
-  std::cout << "INFO: CPU for factorize-solve "
-  	    << float(tend-tstart)/float(CLOCKS_PER_SEC) << std::endl;
-  tstart = tend;
+    tend = clock();
+    std::cout << "INFO: CPU for factorize-solve "
+              << float(tend - tstart) / float(CLOCKS_PER_SEC) << std::endl;
+    tstart = tend;
 
-  unsigned tot_outliers = 0;
-  double old_chi2 = ComputeChi2().chi2;
+    unsigned tot_outliers = 0;
+    double old_chi2 = ComputeChi2().chi2;
 
-  while (true)
+    while (true)
```

Comment:
Yeah, it's probably possible that this doesn't exit. I'll add it to my "jointcal robustness" epic.